nab/detectors/htmjava/src/main/java/nab/detectors/htmjava/HTMModel.java

```diff
@@ -0,0 +1,383 @@
+package nab.detectors.htmjava;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import org.joda.time.DateTimeZone;
+import org.numenta.nupic.Connections;
+import org.numenta.nupic.Parameters;
+import org.numenta.nupic.Parameters.KEY;
+import org.numenta.nupic.algorithms.Anomaly;
+import org.numenta.nupic.algorithms.SpatialPooler;
+import org.numenta.nupic.algorithms.TemporalMemory;
+import org.numenta.nupic.network.Layer;
+import org.numenta.nupic.network.Network;
+import org.numenta.nupic.network.Region;
+import org.numenta.nupic.network.sensor.ObservableSensor;
+import org.numenta.nupic.network.sensor.Publisher;
+import org.numenta.nupic.network.sensor.Sensor;
+import org.numenta.nupic.network.sensor.SensorParams;
+import org.numenta.nupic.util.Tuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HTMModel {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(HTMModel.class);
+
+    private Network network;
+
+    private Publisher publisher;
+
+    /**
+     * Create HTM Model to be used by NAB
+     * @param modelParams OPF Model parameters to parameters from
+     */
+    public HTMModel(JsonNode modelParams) {
+        LOGGER.trace("HTMModel({})", modelParams);
+
+        // Create Sensor publisher to push NAB input data to network
+        publisher = Publisher.builder()
+                .addHeader("timestamp,value")
+                .addHeader("datetime,float")
+                .addHeader("T,B")
+                .build();
+
```

Comment:
Hi Luiz,

As part of the new Persistence API, the idiom for creating the ```Publisher``` has changed. The "new" (or more advanced) way to add a ```PublisherSupplier``` instead.

So you can substitute this code for lines: 45 - 49...

``` Java
PublisherSupplier supplier = PublisherSupplier.builder()
            .addHeader("timestamp,value")
            .addHeader("datetime,float")
            .addHeader("T,B")
            .build();
```
...and then on line 62, add the ```supplier``` instead of the ```publisher```.

That should do it! ;-)